tests/basic/pool_function_test.go

```diff
@@ -932,3 +934,127 @@ var _ = Describe("{PoolExpandTestLimits}", func() {
 		dash.VerifyFatal(err != nil, true, "DMThin pool expansion to 20 TB should result in error")
 	})
 })
+
+var _ = Describe("{PoolExpandAndCheckAlertsUsingAddDisk}", func() {
+
+	var testrailID = 34542894
+	// testrailID corresponds to: https://portworx.testrail.net/index.php?/tests/view/34542894
+
+	BeforeEach(func() {
+		StartTorpedoTest("PoolExpandAndCheckAlertsUsingAddDisk", "pool expansion using add-disk and check alerts after each operation", nil, testrailID)
+		contexts = scheduleApps()
+	})
+	JustBeforeEach(func() {
+		poolIDToResize = pickPoolToResize()
+		log.Infof("Picked pool %s to resize", poolIDToResize)
+		poolToResize = getStoragePool(poolIDToResize)
+	})
+	JustAfterEach(func() {
+		AfterEachTest(contexts)
+	})
+
+	AfterEach(func() {
+		appsValidateAndDestroy(contexts)
+		EndTorpedoTest()
+	})
+
+	It("pool expansion using add-disk and check alerts after each operation", func() {
+		var nodeDetail *node.Node
+		var err error
+
+		stepLog = "Move pool to maintenance mode"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			nodeDetail, err = GetNodeWithGivenPoolID(poolToResize.Uuid)
+			dash.VerifyFatal(err, nil, fmt.Sprintf("Failed to get Node Details using PoolUUID [%v]", poolToResize.Uuid))
+
+			log.InfoD("Bring Node to Maintenance Mode")
+			log.InfoD(fmt.Sprintf("Entering pool maintenance mode on node %s", nodeDetail.Name))
+			err = Inst().V.EnterPoolMaintenance(*nodeDetail)
+			log.FailOnError(err, fmt.Sprintf("fail to enter node %s in maintenance mode", nodeDetail.Name))
```

Comment:
should test wait for PX to come up successfully in maintenance mode before proceeding?

Reply:
Lets use WaitForPoolStatusToUpdate API to wait till PX comes up in maintenance mode.

For eg. as used here: https://github.com/portworx/torpedo/blob/master/tests/common.go#L7564

Reply:
fixed